integration/marathon_test.go

```diff
@@ -1,42 +1,69 @@
 package integration
 
 import (
+	"fmt"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/containous/traefik/integration/try"
+	"github.com/containous/traefik/types"
+	marathon "github.com/gambol99/go-marathon"
 	"github.com/go-check/check"
 	checker "github.com/vdemeester/shakers"
 )
 
+const (
+	containerNameMesosSlave = "mesos-slave"
+	containerNameMarathon   = "marathon"
+)
+
 // Marathon test suites (using libcompose)
 type MarathonSuite struct{ BaseSuite }
 
 func (s *MarathonSuite) SetUpSuite(c *check.C) {
 	s.createComposeProject(c, "marathon")
 	s.composeProject.Start(c)
+	// Add entry for Mesos slave container IP address in the hosts file so
+	// that Traefik can properly forward traffic.
+	// This is necessary as long as we are still using the docker-compose v1
+	// spec. Once we switch to v2 or higher, we can have both the test/builder
+	// container and the Mesos slave container join the same custom network and
+	// enjoy DNS-discoverable container host names.
+	mesosSlaveIPAddr := s.getContainerIPAddr(c, containerNameMesosSlave)
+	c.Assert(s.extendDockerHostsFile(containerNameMesosSlave, mesosSlaveIPAddr), checker.IsNil)
+}
 
-	// FIXME Doesn't work...
-	//// "github.com/gambol99/go-marathon"
-	//config := marathon.NewDefaultConfig()
-	//
-	//marathonClient, err := marathon.NewClient(config)
-	//if err != nil {
-	//	c.Fatalf("Error creating Marathon client. %v", err)
-	//}
-	//
-	//// Wait for Marathon to elect itself leader
-	//err = try.Do(30*time.Second, func() error {
-	//	leader, err := marathonClient.Leader()
-	//
-	//	if err != nil || len(leader) == 0 {
-	//		return fmt.Errorf("Leader not found. %v", err)
-	//	}
-	//
-	//	return nil
-	//})
-	//
-	//c.Assert(err, checker.IsNil)
+func (s *MarathonSuite) getContainerIPAddr(c *check.C, name string) string {
```

Comment:
I can move it if you like; just to be sure though: did you notice that I need it once for the Mesos slave and another time for Marathon? That'd be 4 lines, so I wanted to avoid code duplication. (In fact, the function would be reusable if we turned `s.composeProject` into a function parameter, which would allow us to apply it in a lot of other tests as well.)

Personally, I also find `getContainerIPAddr(c, name)` more readable and comprehendible compared to `s.composeProject.Container(c, name).NetworkSettings.IPAddress`.